main.c

```diff
@@ -2346,6 +2267,13 @@ ovl_do_open (fuse_req_t req, fuse_ino_t parent, const char *name, int flags, mod
           return -1;
         }
 
+      sprintf (path, "%s/%s", p->path, name);
```

Comment:
Should this be snprintf?

Reply:
good idea!  Now that we have autocleanup, we should use only dynamic allocated buffers.

I've added another patch to change all static buffers to dynamic allocated ones.

Please do not merge yet, I want to do some more testing as it is a lot of code changed, I will mark the PR as WIP.  If you are fine with the changes, I can merge after the stress tests

Reply:
SGTM